packages/web-components/@groupby/elements-base/src/base.ts

```diff
@@ -15,3 +20,30 @@ export default abstract class Base extends LitElement {
     return this.dispatchEvent(eventToDispatch);
   }
 }
+
+/**
+ * A property decorator that will set the data initialization state in a class.
+ * It will extend the property's setter to set the initialization state to true.
+ * This expects that the property already has defined accessors and a default value.
+ *
+ * @param initProperty The name of the initialization property. Has a default value
+ * of "_initialized".
+ */
+export function dataInitializer(initProperty: string = '_initialized'): PropertyDecorator {
```

Comment:
This is a separate entity and not coupled to `Base`, so I recommend splitting this out into a separate file.

Reply:
Moved to a `decorators.ts` file in the same directory.